### ISSUE TITLE
K8s conformance test failure: [sig-scheduling] SchedulerPredicates [Serial] validates that there is no conflict between pods with same hostPort but different hostIP and protocol

### ISSUE THREAD

**Issue by etungsten** [projectcalico/cni-plugin#1213]
## Expected Behavior

K8s conformance tests all pass for K8s v1.20.x and 1.21.x with calico cni v3.21

## Current Behavior

With Calico CNI v3.21 one test consistently fails:
```
[sig-network] HostPort validates that there is no conflict between pods with same hostPort but different hostIP and protocol [LinuxOnly] [Conformance]
```
Test log:
https://gist.github.com/etungsten/2317de8cb4cfe989aba388553f4b137d

## Possible Solution

I'm unsure what the exact cause is.
Something changed in v3.21 that is causing this regression for us.

## Steps to Reproduce (for bugs)

1. Provision K8s clusters in vSphere with K8s cluster-api
2. Deploy calico CNI version 3.21 manifests in newly provisioned clusters.
3. Try running sonobuoy K8s conformance test suite on the clusters with Bottlerocket nodes.
4. All tests pass except for `[sig-network] HostPort validates that there is no conflict between pods with same hostPort but different hostIP and protocol [LinuxOnly] [Conformance]`

## Context

We're unable to validate new Bottlerocket OS image releases when testing in clusters with the latest (v3.21) calico cni plugin.
Temporarily working around by pinning to v3.20 of the CNI plugin.

## Your Environment

* CNI plugin version (e.g. `/opt/cni/bin/calico --version`): 3.21
* Orchestrator version (e.g. kubernetes, mesos, rkt): k8s v1.20.5, v1.20.6, v1.21.11, v1.21.12, possibly others
* CNI config: https://docs.projectcalico.org/v3.21/manifests/calico.yaml
* Operating System and version: Bottlerocket vmware-k8s-1.20 v1.3.0 
* Link to your project (optional): https://github.com/bottlerocket-os/bottlerocket

**Comment by caseydavenport** [comment]
Thanks for the issue - we've fixed this and are working on a v3.21.1 patch release to address it. Should be out soon.

**Comment by caseydavenport** [comment]
This was the fix: https://github.com/projectcalico/libcalico-go/pull/1544
